.github/workflows/lib-injection.yml

```diff
@@ -44,3 +44,51 @@ jobs:
           docker-registry-username: ${{ github.repository_owner }}
           docker-registry-password: ${{ secrets.GITHUB_TOKEN }}
           test-script: ./lib-injection/run-manual-lib-injection.sh
+
+  test_unit:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        variant: [
+          'dd-lib-python-init-test-django',
+          'dd-lib-python-init-test-django-gunicorn',
+          'dd-lib-python-init-test-django-uvicorn',
+        ]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Create a docker network for the app and test agent
+        run: |
+          docker network create test-inject
+      - name: Run the test agent
+        run: |
+          docker run \
+            -d \
+            --network=test-inject \
+            -p 8126:8126 \
+            ghcr.io/datadog/dd-apm-test-agent/ddapm-test-agent:v1.7.2
+      - name: Apply a fixed, stable version of the library
+        run: sed -i "s~<DD_TRACE_VERSION_TO_BE_REPLACED>~1.8.0~g" lib-injection/sitecustomize.py
```

Comment:
Is this idea to test this on every PR? shouldn't we be testing each branch?

Reply:
Hmm my thinking here was that we didn't need to test each branch as we just want to test the mechanism but it's easy enough to test with the current branch. I'll make that change :+1: